classes.py

```diff
@@ -123,116 +130,356 @@ def from_dict(data):
+        cls.save_show(vars(show))
+
+    @classmethod
+    def edit_show(cls, show_id, new_event_id, new_start_time, new_end_time,
+                  new_price, new_datetime):
+        cls.delete_show(show_id)
+        show = cls(show_id, new_event_id, new_start_time, new_end_time,
+                      new_price, new_datetime)
+        cls.save_show(vars(show))
+
+    def save_show(show: dict) -> None:
+        dic = get_list_from_json()
+        show_id = Show._get_next_id()
+        dic.update({show_id: show})
+        try:
+            with open("jsons/show.json", "w") as fp:
+                json.dump(dic, fp, indent=4)  # encode dict into JSON
+        except Exception as ex:
+            print('You have error', ex)
+
+    def delete_show(show_id: str) -> None:
+        dic = get_list_from_json()
+        del dic[show_id]
+        try:
+            with open("jsons/show.json", "w") as fp:
+                json.dump(dic, fp, indent=4)  # encode dict into JSON
+        except Exception as ex:
+            print('You have error', ex)
+
+
+    @staticmethod
+    def show_which_show(event_id):
+        show = list((get_list_from_json).values())
+        for show in show:
+            if show['event_id'] == event_id:
+                show = get_list_from_json(show['show_id'])
+                return f"({show['show_id']}) - |{show['start_time']} to {show['end_time']} \n      "\
+                       f"|price : {show['price']}"
+            else:
+                raise ValueError('not found show for this event')
+
+
 class Ticket:
-    """Daje Id biletom"""
-    tickets = []
-    for i in tickets:
-        tickets.append(i)
+    _id_counter = 0
 
-    def __init__(
-            self,
-            id: int,
-            event_name: str,
-            participant_id: int,
-            row: str,
-            place: str
-    ):
-        """Pojedynczy bilet"""
-        super().__init__(self)
-        self._id = id  # zamienić na generator identyfikatorow, zeby nie bylo duplikatow
-        self.event_name = event_name  # relacja do wydarzenia
-        self.participant_id = participant_id  # relacja do uczestnika
-        self.row = row
-        self.place = place
+    @classmethod
+    def _get_next_id(cls):
+        cls._id_counter += 1
+        return cls._id_counter
 
-    def to_dict(self):
-        result = vars(self).copy()  # Użyjemy kopii, aby nie modyfikować oryginalnego słownika
-        for key, value in result.items():
-            if isinstance(value, datetime):
-                result[key] = value.isoformat()
-        return result
+    @classmethod
+    def set_id_counter(cls, new_max_id):
+        cls._id_counter = new_max_id
 
-class Participant(Person):
-    """Uczestnik wydarzenia"""
+    def __init__(self, ticket_id, show_id, participant_id):
+        self.ticket_id = Ticket._get_next_id(), ticket_id
+        self.show_id = Show._get_next_id(), show_id
+        self.participant_id = Participant._get_next_id(), participant_id
```

Comment:
O co tu chodzi z tymi krotkami? czemu dajesz najpierw generator, a potem po przecinku parametr?

Reply:
parametr był stosowany pierwotnie, kiedy nie był jeszcze stworzony generator. Większość funkcji mam spiętych między sobą.  Trzeba przerabiać kod.

Reply:
no to go przerób i zrób nowy commit i push

Reply:
![image](https://github.com/infoshareacademy/jpydzr5-apocalypse-programmers/assets/5888903/d157d9dc-e8f6-4b60-a209-d507923a5075)
Da się wstawiać zrzuty